### ISSUE TITLE
Discarding a shared task doesn't work

### ISSUE THREAD

**Issue by ddisanti** [mpcjanssen/simpletask-android#638]
If you share a task with Simpletask and for some reason you don't want to save it, taping on the upper left ```X``` button  doesn't work as expected. The task is saved anyway.

**Comment by ddisanti** [comment]
Oh. I just get it. The ```X``` is for to discard any changes you may did to the shared task. Right?
Anyway, there should be an option to discard the task, ie to not save it to the list.

**Comment by smichel17** [comment]
How Simpletask works right now:

- Tasks each have a state: selected or not selected
- Editing a task and adding a new task both work in the exact same way:
- When you launch edit/add screen, you don't tell it anything about why you're opening it
- The app edit/add screen knows what you want it to do based on the current state of the app (are the tasks selected? Then it's an edit. What filter is open? We can pre-fill with those lists/tags.)

So, the way the `Edit shared tasks` setting works behind the scenes is:

1. Add shared tasks to the task list
2. Select those tasks
3. Launch the add/edit screen

That's why this bug exists.

The way it should work is that each time we launch that screen we should tell it a little bit of information about which tasks we actually want to add/edit. Then we should treat 'edit before sharing' as adding new tasks, instead of editing existing ones.

This is also the proper way to fix #547 (the `easy` label is because there is also a quick fix of just making sure no tasks are selected, when want to add a new one).

**Comment by ddisanti** [comment]
Maybe, a dirty fix would be make the ```X``` button to delete the task.
Anyway, this happened to me because sometimes I share things to Simpletask by mistake and don't want it to save it.  I have Simpletask on top of my _"Share with..."_ menu.

**Comment by smichel17** [comment]
Wouldn't work. AddTask can't distinguish between tasks shared to simpletask and then edited and when you select tasks on the main screen and then hit 'edit'.

**Comment by ddisanti** [comment]
Ok. It is just I though that if the task is identified, then selected and finally loaded to the _Edit screen_  it would be easy to delete if you want it to.
@smichel17 Anyway, is not a big deal.
@mpcjanssen you can close this if you think this can't be implemented or it will be of no use.

**Comment by smichel17** [comment]
I do not think we should close it. Hitting the X after editing shared tasks should discard those tasks, not add them. It's not a high priority, but that's fine; it should still be here.

**Comment by t214c** [comment]
Is there anyway this could be fixed?

**Comment by mpcjanssen** [comment]
Main goal is to prevent tasks from being lost (for example if you add a background task and forget to apply the edit box). This behavior is a side effect of this. I think in this case, not losing any tasks trumps having the occasional additional task.
Properly fixing this requires distinguishing the different edit scenarios which is quite a bit of work. For now a workaround is to clear the edit box, which should discard the tasks.

**Comment by t214c** [comment]
OK. Thanks.

**Comment by ddisanti** [comment]
> For now a workaround is to clear the edit box, which should discard the tasks.

@mpcjanssen I've tried that back then.  As @smichel17 said, the task is already saved. Clearing the Edit box doesn't work. It does not work by Clearing the Edit Box and then taping the Save button, or by pressing the X Button or pressing the Back button. The task is saved anyway.
